dropshot_endpoint/src/endpoint.rs

```diff
@@ -384,34 +384,11 @@ impl<'ast> EndpointParams<'ast> {
         let ret_ty = self.ret_ty;
         let ret_check = quote_spanned! { ret_ty.span()=>
             const _: fn() = || {
-                trait ResultTrait {
-                    type T;
-                    type E;
-                }
-                impl<TT, EE> ResultTrait for Result<TT, EE>
-                where
-                    TT: #dropshot::HttpResponse,
-                {
-                    type T = TT;
-                    type E = EE;
-                }
-                struct NeedHttpResponse(
-                    <#ret_ty as ResultTrait>::T,
-                );
-                trait TypeEq {
-                    type This: ?Sized;
-                }
-                impl<T: ?Sized> TypeEq for T {
-                    type This = Self;
-                }
-                fn validate_result_error_type<T>()
+                fn validate_response_type<T>()
```

Comment:
all of this is extraneous? anything relevant here to juice up the error messages?